ecs-agent/netlib/platform/common_linux.go

```diff
@@ -0,0 +1,200 @@
+//go:build !windows
+// +build !windows
+
+package platform
+
+import (
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/ecs_client/model/ecs"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/ecscni"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/networkinterface"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/tasknetworkconfig"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/pkg/errors"
+)
+
+const (
+	// Identifiers for each platform we support.
+	WarmpoolDebugPlatform    = "ec2-debug-warmpool"
+	FirecrackerDebugPlatform = "ec2-debug-firecracker"
+	WarmpoolPlatform         = "warmpool"
+	FirecrackerPlatform      = "firecracker"
+)
+
+// common will be embedded within every implementation of the platform API.
+// It contains all fields and methods that can be commonly used by all
+// platforms.
+type common struct {
+	nsUtil ecscni.NetNSUtil
+}
+
+// NewPlatform creates an implementation of the platform API depending on the
+// platform type where the agent is executing.
+func NewPlatform(
+	platformString string,
+	nsUtil ecscni.NetNSUtil) (API, error) {
+	commonPlatform := common{
+		nsUtil: nsUtil,
+	}
+
+	switch platformString {
+	case WarmpoolPlatform:
+		return &containerd{
+			common: commonPlatform,
+		}, nil
+	}
```

Comment:
Can we add a TODO for the missing platforms (so we can see our way toward EC2 integration)?

Reply:
Will do.